cli/src/main/kotlin/commands/RequirementsCommand.kt

```diff
@@ -0,0 +1,101 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.commands
+
+import ch.frankel.slf4k.*
+
+import com.beust.jcommander.JCommander
+import com.beust.jcommander.Parameters
+
+import com.here.ort.CommandWithHelp
+import com.here.ort.analyzer.PackageManager
+import com.here.ort.model.config.AnalyzerConfiguration
+import com.here.ort.model.config.RepositoryConfiguration
+import com.here.ort.model.config.ScannerConfiguration
+import com.here.ort.scanner.Scanner
+import com.here.ort.utils.CommandLineTool
+import com.here.ort.utils.log
+
+import com.vdurmont.semver4j.Requirement
+
+import org.reflections.Reflections
+
+import java.lang.reflect.Modifier
+
+@Parameters(commandNames = ["requirements"], commandDescription = "List the required command line tools.")
+object RequirementsCommand : CommandWithHelp() {
+    override fun runCommand(jc: JCommander) {
+        val reflections = Reflections("com.here.ort")
+        val classes = reflections.getSubTypesOf(CommandLineTool::class.java)
+
+        val allTools = mutableMapOf<String, MutableList<CommandLineTool>>()
+        val anyVersion = Requirement.buildNPM("*").toString()
+
+        classes.filterNot {
+            Modifier.isAbstract(it.modifiers) || it.isAnonymousClass
+        }.sortedBy { it.simpleName }.forEach {
+            try {
+                val kotlinObject = it.kotlin.objectInstance
+
+                var key = "Other tool"
+                val instance = when {
+                    kotlinObject != null -> {
+                        log.debug { "$it is a Kotlin object." }
+                        kotlinObject
+                    }
+
+                    PackageManager::class.java.isAssignableFrom(it) -> {
+                        key = "PackageManager"
+                        log.debug { "$it is a $key." }
+                        it.getDeclaredConstructor(AnalyzerConfiguration::class.java,
+                                RepositoryConfiguration::class.java).newInstance(AnalyzerConfiguration(false, false),
+                                RepositoryConfiguration(null))
+                    }
+
+                    Scanner::class.java.isAssignableFrom(it) -> {
+                        key = "Scanner"
+                        log.debug { "$it is a $key." }
+                        it.getDeclaredConstructor(ScannerConfiguration::class.java)
+                                .newInstance(ScannerConfiguration())
+                    }
+
+                    else -> {
+                        log.debug { "Trying to instanciate $it without any arguments." }
+                        it.getDeclaredConstructor().newInstance()
+                    }
+                }
+
+                if (instance.command().isNotEmpty() && instance.getVersionRequirement().toString() != anyVersion) {
```

Comment:
I think we should also list tools which have no version requirement, because they still require that the tool is installed.

Reply:
True, done.